addons/resource/models/resource.py

```diff
@@ -944,16 +944,16 @@ def _adjust_to_calendar(self, start, end):
         end, revert_end_tz = make_aware(end)
         result = {}
         for resource in self:
-            calendar_start = resource.calendar_id._get_closest_work_time(start, resource=resource)
-            search_range = None
             tz = timezone(resource.tz)
-            if calendar_start and start.astimezone(tz).date() == end.astimezone(tz).date():
```

Comment:
For a user in Belgium:
start = day-1, 22:00
end = day, 21:59:59
tz is set at the resource tz (assumed to be Asia/Calcutta)
=>
start.astimezone(tz): day-1, 22:00 => day, 3:30
end.astimezone(tz): day, 21:59:59 => day+1, 2:29:59
The .date() of the two datetimes will not be equal and the search_range won't be assigned the value of:
search_range = (start, end + relativedelta(days=1, hour=0, minute=0, second=0)